vitrivr-engine-module-pgvector/src/main/kotlin/org/vitrivr/engine/database/pgvector/descriptor/AbstractDescriptorReader.kt

```diff
@@ -0,0 +1,225 @@
+package org.vitrivr.engine.database.pgvector.descriptor
+
+import org.vitrivr.engine.core.database.descriptor.DescriptorReader
+import org.vitrivr.engine.core.model.descriptor.Descriptor
+import org.vitrivr.engine.core.model.descriptor.DescriptorId
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.query.Query
+import org.vitrivr.engine.core.model.retrievable.RetrievableId
+import org.vitrivr.engine.core.model.retrievable.Retrieved
+import org.vitrivr.engine.database.pgvector.*
+import java.sql.ResultSet
+import java.util.*
+
+/**
+ * An abstract implementation of a [DescriptorReader] for Cottontail DB.
+ *
+ * @author Ralph Gasser
+ * @version 1.0.0
+ */
+abstract class AbstractDescriptorReader<D : Descriptor>(final override val field: Schema.Field<*, D>, override val connection: PgVectorConnection) : DescriptorReader<D> {
+
+    /** The name of the table backing this [AbstractDescriptorReader]. */
+    protected val tableName: String = "${DESCRIPTOR_ENTITY_PREFIX}_${this.field.fieldName}"
+
+
+    /** The [Descriptor] prototype handled by this [AbstractDescriptorReader]. */
+    protected val prototype: D= this.field.analyser.prototype(this.field)
+
+    /**
+     * Returns a single [Descriptor]s of type [D] that has the provided [DescriptorId].
+     *
+     * @param descriptorId The [DescriptorId], i.e., the [UUID] of the [Descriptor] to return.
+     * @return [Sequence] of all [Descriptor]s.
+     */
+    override fun get(descriptorId: DescriptorId): D? {
+        try {
+            this.connection.jdbc.prepareStatement("SELECT * FROM $tableName WHERE $DESCRIPTOR_ID_COLUMN_NAME = ?").use { stmt ->
+                stmt.setObject(1, descriptorId)
+                stmt.executeQuery().use { res ->
+                    if (res.next()) {
+                        return this.rowToDescriptor(res)
+                    } else {
+                        return null
+                    }
+                }
+            }
+        } catch (e: Exception) {
+            LOGGER.error(e) { "Failed to check fetch descriptor $descriptorId from '$tableName' due to SQL error." }
+            return null
+        }
+    }
+
+    /**
+     * Returns the [Descriptor]s of type [D] that belong to the provided [RetrievableId].
+     *
+     * @param retrievableId The [RetrievableId] to search for.
+     * @return [Sequence] of [Descriptor]  of type [D]
+     */
+    override fun getFor(retrievableId: RetrievableId): Sequence<D> {
```

Comment:
I don't know how this changed, but the original idea behind `getFor` was to have two parameters to be able to specify the semantics of the supplied id. If this is now exclusively used for retrievable ids, it should be renamed.

Reply:
Since there is no other notion of an ID on `Descriptor`s i think it makes sense to settle for this option. I'd argue that performing a lookup on any other field is simply a boolean query,

What name would you propose?

Reply:
Fair enough. Since this is then explicitly a mechanism to get the descriptor for a retrievable id, maybe just `forRetrievable(...)`?

Reply:
Done.